paddle/fluid/operators/mkldnn/fill_zeros_like_mkldnn_op.cc

```diff
@@ -0,0 +1,85 @@
+/* Copyright (c) 2021 PaddlePaddle Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License. */
+
+#include "paddle/fluid/platform/mkldnn_reuse.h"
+
+namespace paddle {
+namespace operators {
+
+using framework::Tensor;
+
+template <typename T>
+class FillZerosLikeMKLDNNHandler
+    : public platform::MKLDNNHandlerNoCachingT<T, dnnl::eltwise_forward,
+                                               dnnl::eltwise_backward> {
+ public:
+  FillZerosLikeMKLDNNHandler(Tensor* out, const dnnl::engine engine,
+                             platform::Place cpu_place)
+      : platform::MKLDNNHandlerNoCachingT<T, dnnl::eltwise_forward,
+                                          dnnl::eltwise_backward>(engine,
+                                                                  cpu_place) {
+    // uint8_t is always used to treat float NaNs like normal numbers
+    auto md = dnnl::memory::desc(
+        {out->numel() * static_cast<int64_t>(sizeof(T))},
+        dnnl::memory::data_type::u8, dnnl::memory::format_tag::a);
+
+    this->AcquireForwardPrimitiveDescriptor(dnnl::prop_kind::forward_training,
+                                            dnnl::algorithm::eltwise_linear, md,
+                                            0.0f, 0.0f);

```

Comment:
Are we sure that this oneDNN kernel will be faster than native kernel? Or we have to wait for real models to check ?

Reply:
We have to wait and check